taxonomy/utils.py

```diff
@@ -662,3 +664,33 @@ def duplicate_xblock_skills(source_xblock_uuid, xblock_uuid):
     for source_xblock_skill in source_xblock_skills:
         source_xblock_skill.xblock = xblock
         duplicate_model_instance(source_xblock_skill)
+
+
+def update_xblock_skills_verification_counts(usage_key: str, verified_skills: List[int], ignored_skills: List[int]):
+    """
+    Update xblock skill verified and ignored count.
+
+    Arguments:
+        usage_key (str): usage_key of xblock.
+        verified_skills (List[int]): list of verified skill ids.
+        ignored_skills (List[int]): list of ignored skill ids.
+    """
+    xblock_model, identifier = get_product_skill_model_and_identifier(ProductTypes.XBlock)
+    try:
+        xblock = xblock_model.objects.get(**{identifier: usage_key})
+    except ObjectDoesNotExist:
+        LOGGER.error(f'[TAXONOMY] XBlock with usage_key: {usage_key} not found!')
+        return
+    xblock_data_model, _ = get_product_skill_model_and_identifier(ProductTypes.XBlockData)
+    # update verified_count
+    xblock_data_model.objects.filter(
+        xblock=xblock,
+        skill_id__in=verified_skills,
+        is_blacklisted=False,
+    ).update(verified_count=F('verified_count') + 1)
+    # update ignored_count
+    xblock_data_model.objects.filter(
+        xblock=xblock,
+        skill_id__in=ignored_skills,
+        is_blacklisted=False,
+    ).update(ignored_count=F('ignored_count') + 1)
```

Comment:
Question: Can you explain why are we adding +1 to both counts here?

Reply:
@sameenfatima78  `ignored_skills` contains ids of skills which were ignored and `verified_skills` contain ids of verified skills. So we just take them and increment their counts. If it is empty nothing will be updated.

Reply:
It's a bit confusing that on `XBLOCK_SKILL_VERIFIED` signal handler we are updating the `ignored_count`. I'm assuming this signal will be called when an `xblock` is verified, is this correct?

Reply:
[XBLOCK_SKILL_VERIFIED](https://github.com/openedx/openedx-events/pull/165/files#diff-5cf3a0b26961626ce777acf38a3cd803cb0ab982f8f35a35d078cb9c62438fa3R223-R235) is sending both verified and ignored skill ids as there is only one action to submit this information.

![SS](https://user-images.githubusercontent.com/10894099/210078679-3cbac3d1-55a7-4fba-b841-7fb4468f32c5.png)

Reply:
Ah okay, got it. Thanks for clarifying.